libbeat/kibana/transform.go

```diff
@@ -7,46 +7,62 @@ import (
 	"github.com/elastic/beats/libbeat/common"
 )
 
-func TransformFields(timeFieldName string, title string, commonFields common.Fields) common.MapStr {
-	fields := []common.MapStr{}
-	fieldFormatMap := common.MapStr{}
-	keys := common.MapStr{}
+type Transformer struct {
+	fields                    common.Fields
+	transformedFields         []common.MapStr
+	transformedFieldFormatMap common.MapStr
+	timeFieldName             string
+	title                     string
+	keys                      common.MapStr
+}
+
+func NewTransformer(timeFieldName, title string, fields common.Fields) Transformer {
```

Comment:
Same comment as above that we could return a reference. Didn't check if it is required in this case. It seems to work just fine :-)